pkg/providers/v1/aws.go

```diff
@@ -1156,6 +1164,15 @@ func (s *awsSdkEC2) DescribeSubnets(request *ec2.DescribeSubnetsInput) ([]*ec2.S
 	return response.Subnets, nil
 }
 
+func (s *awsSdkEC2) DescribeAvailabilityZones(request *ec2.DescribeAvailabilityZonesInput) ([]*ec2.AvailabilityZone, error) {
+	// AZs are not paged
+	response, err := s.ec2.DescribeAvailabilityZones(request)
```

Comment:
You may also need to update the [IAM Policy document](https://github.com/kubernetes/cloud-provider-aws/blob/master/docs/prerequisites.md#iam-policies) by adding the `ec2:DescribeAvailabilityZones` permission. That IAM Policy should be attached to the control plane IAM Instance Profile/Role.

Reply:
Updated